demisto_sdk/commands/common/tests/integration_test.py

```diff
@@ -411,51 +413,55 @@ def test_invalid_feed(self, feed, fromversion):
-        with open(validator.file_path, 'r') as f:
-            _file = safe_load(f)
-        _file['configuration'] = [t for t in _file['configuration'] if not t['name'] == 'incidentType']
-        validator.current_file = _file
-
-        assert validator.is_fetch_params_exist() is False, 'is_fetch_params_exist() returns True instead False'
-
-    def _test_is_fetch_params_exist_missing_field(self):
-        # missing param
-        structure = mock_structure("")
-        validator = IntegrationValidator(structure)
-        validator.file_path = 'demisto_sdk/tests/test_files/Akamai_SIEM.yml'
-        with open(validator.file_path, 'r') as f:
-            _file = safe_load(f)
-        _file['configuration'] = [t if not t['name'] == 'incidentType' else t.pop('name')
-                                  for t in _file['configuration']]
-        validator.current_file = _file
-        assert validator.is_fetch_params_exist() is False, 'is_fetch_params_exist() returns True instead False'
-
-    def test_is_fetch_params_exist_malformed_field(self):
-        # incorrect param
-        structure = mock_structure("")
-        validator = IntegrationValidator(structure)
-        validator.file_path = 'demisto_sdk/tests/test_files/Akamai_SIEM.yml'
-        with open(validator.file_path, 'r') as f:
-            _file = safe_load(f)
-        validator.current_file = _file
-        validator.current_file['configuration'] = []
-        for t in _file['configuration']:
-            if t.get('name') == 'incidentType':
-                t['type'] = 123
-            validator.current_file['configuration'].append(t)
-
-        assert validator.is_fetch_params_exist() is False, 'is_fetch_params_exist() returns True instead False'
+    class TestIsFetchParamsExist:
+        config = {
+            'configuration': FETCH_REQUIRED_PARAMS,
+            'script': {'isfetch': True}
+        }
+        validator = IntegrationValidator(mock_structure("", config))
+
+        def setup(self):
+            return copy(self.validator)
+
+        def test_valid(self):
+            validator = self.setup()
+            assert validator.is_fetch_params_exist(), 'is_fetch_params_exist() returns False instead True'
+
+        def test_sanity(self):
+            # missing param in configuration
+            validator = self.setup()
+            validator.current_file['configuration'] = [t for t in validator.current_file['configuration']
+                                                       if t['name'] != 'incidentType']
+            assert validator.is_fetch_params_exist() is False, 'is_fetch_params_exist() returns True instead False'
+
+        def test_missing_field(self):
+            # missing param
+            validator = self.setup()
+            configuration = validator.current_file['configuration']
+            for i in range(len(configuration)):
+                if configuration[i]['name'] == 'incidentType':
+                    del configuration[i]['name']
```

Comment:
you can't remove elements from the list while going over it
also, use `enumerate` instead of `range(len(<list>))`